src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/attribute_editor.cljs

```diff
@@ -1,121 +1,95 @@
-                      :payload del-ops
-                      :headers utils/content-type=json
-                      :on-done (fn [{:keys [success? xhr]}]
-                                 (swap! state dissoc :deleting-attrs?)
-                                 (if-not success?
-                                   (do
-                                     (js/alert (str "Exception:\n" (.-statusText xhr)))
-                                     (swap! state assoc :attrs-list (:orig-attrs @state))
-                                     (swap! state dissoc :orig-attrs)
-                                     (react/call :load-workspace this))
-                                   (when-not (empty? up-map)
-                                     (update-orch-fn up-map))))}))]
-    (cond
-      (some empty? curr-keys) (js/alert "Empty attribute keys are not allowed!")
-      (some empty? curr-vals) (js/alert "Empty attribute values are not allowed!")
-      (not (or (empty? curr-keys) (apply distinct? curr-keys))) (js/alert "Unique keys must be used!")
-      :else (do
-              (if (empty? del-map)
-                (when-not (empty? up-map)
-                  (update-orch-fn up-map))
-                (del-orch-fn del-map))
-              (swap! state update-in [:server-response :workspace "workspace" "attributes"] assoc "description" description)
-              (swap! state assoc :editing? false)))))
 
+(react/defc WorkspaceAttributeViewerEditor
+  {:get-attributes
+   (fn [{:keys [state]}]
+     (let [{:keys [attributes]} @state
+           duplicates (not-empty (utils/find-duplicates (map key attributes)))
+           any-empty? (some (fn [[k v]]
+                              (let [[ek ev] (map (comp empty? trim) [k v])]
+                                (or ek ev)))
+                            attributes)
+           with-spaces (->> attributes
+                            (map (comp trim key))
+                            (filter (partial re-find #"\s"))
+                            not-empty)]
+       (cond duplicates {:error (str "Duplicate keys: " (join ", " duplicates))}
+             any-empty? {:error "Empty keys and values are not allowed."}
+             with-spaces {:error (str "Keys cannot have spaces: " (join ", " with-spaces))}
+             :else {:success (into {} attributes)})))
+   :render
+   (fn [{:keys [props state after-update]}]
+     (let [{:keys [editing?]} props]
+       [:div {}
+        (style/create-section-header "Workspace Attributes")
+        (style/create-paragraph
+          [:div {}
+           (when editing?
+             [:div {:style {:marginBottom "0.25em"}}
+              [comps/Button {:icon :add :text "Add new"
+                             :onClick (fn [_]
+                                        (swap! state update :attributes conj ["" ""])
+                                        ;; have to do this by ID not ref, since the fields are generated within Table
```

Comment:
Good comment, good hack. Let's chat offline about a strategy to make this more "Reacty."